docs/docs/state-and-lifecycle.md

```diff
@@ -0,0 +1,438 @@
+
+The `Clock` state is now encapsulated. However it doesn't update yet.
+
+Next, we'll make the `Clock` update its own state every second.
+
+## Adding Lifecycle Methods to a Class
+
+In applications with many components, it's very important to free up resources taken by the components when they are destroyed.
+
+We want to [set up an interval](https://developer.mozilla.org/en-US/docs/Web/API/WindowTimers/setInterval) whenever the `Clock` is rendered to the DOM for the first time. This is called "mounting" in React.
+
+We also want to [clear that interval](https://developer.mozilla.org/en-US/docs/Web/API/WindowTimers/clearInterval) whenever the DOM produced by the `Clock` is removed. This is called "unmounting" in React.
+
+We can declare special methods on the component class to run some code when a component mounts and unmounts:
+
+```js{7-9,11-13}
+class Clock extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {date: new Date()};
+  }
+
+  componentDidMount() {
+
+  }
+
+  componentWillUnmount() {
+
+  }
+
+  render() {
+    return (
+      <div>
+        <h1>Hello, world!</h1>
+        <h2>It is {this.state.date.toLocaleTimeString()}.</h2>
+      </div>
+    );
+  }
+}
+```
+
+These methods are called "lifecycle hooks".
+
+The `componentDidMount()` hook runs after the component output has been rendered to the DOM. This is a good place to set up an interval:
+
+```js{2}
+  componentDidMount() {
+    this.intervalId = setInterval(this.tick, 1000);
+  }
+```
+
+Note how we save the interval ID right on `this`. While `this.props` is set up by React itself and `this.state` has a special meaning, you are free to add additional fields to the class manually if you need to store something that is not used for the visual output, such as the timeout ID.
+
+We will tear down the interval in the `componentWillUnmount()` lifecycle hook:
+
+```js{2}
+  componentDidMount() {
+    clearInterval(this.intervalId);
+  }
+```
+
+Finally, we will implement the `tick()` method that runs every second.
+
+It will use `this.setState()` to schedule updates to the component local state:
+
+```js{4,16-20}
+class Clock extends React.Component {
+  constructor(props) {
+    super(props);
+    this.tick = this.tick.bind(this);
+    this.state = {date: new Date()};
+  }
+
+  componentDidMount() {
+    this.intervalId = setInterval(this.tick, 1000);
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.intervalId);
+  }
+
+  tick() {
+    this.setState({
+      date: new Date()
+    });
+  }
+
+  render() {
+    return (
+      <div>
+        <h1>Hello, world!</h1>
+        <h2>It is {this.state.date.toLocaleTimeString()}.</h2>
+      </div>
+    );
+  }
+}
+
+const container = document.getElementById('root');
+ReactDOM.render(<Clock />, container);
+```
+
+[Try it on Codepen.](http://codepen.io/gaearon/pen/amqdNA?editors=0010)
+
+Now the clock ticks every second.
+
+## A Note on Binding Methods
```

Comment:
We could delay talking about binding until the "handling events" doc. I think in practice when you start to handle events is going to be when people first run into binding functions.